File: Domains/Apps/Workspace/Typescript/Intranet/src/allors/domain/apps/TelecommunicationsNumber.ts
import { domain } from "../domain";
import { TelecommunicationsNumber } from "../generated/TelecommunicationsNumber.g";

declare module "../generated/TelecommunicationsNumber.g" {
  interface TelecommunicationsNumber {
    displayName;
  }
}

domain.extend((workspace) => {

  const obj: TelecommunicationsNumber = workspace.prototypeByName["TelecommunicationsNumber"];

  Object.defineProperty(obj, "displayName", {
    get(this: TelecommunicationsNumber) {
      let numberString: string = "";
      if (this.CountryCode || this.AreaCode) {
        if (this.CountryCode && this.AreaCode) {
          numberString = this.CountryCode + " " + this.AreaCode;
        } else if (this.CountryCode) {
          numberString = this.CountryCode;
        } else {
          numberString = this.AreaCode;
        }
      }

      if (numberString === "" && this.ContactNumber) {
        return numberString = this.ContactNumber;
      } else {
        return numberString += " " + this.ContactNumber;
      }
    },
  });

});

File: Base/Database/Domain/Export/Base/Relation/TelecommunicationsNumber.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="TelecommunicationsNumber.cs" company="Allors bvba">
//   Copyright 2002-2012 Allors bvba.
// Dual Licensed under
//   a) the General Public Licence v3 (GPL)
//   b) the Allors License
// The GPL License is included in the file gpl.txt.
// The Allors License is an addendum to your contract.
// Allors Applications is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
// For more information visit http://www.allors.com/legal
// </copyright>
// --------------------------------------------------------------------------------------------------------------------
namespace Allors.Domain
{
    public partial class TelecommunicationsNumber
    {
        public bool IsPostalAddress => false;

        public override string ToString()
        {
            var numberString = string.Empty;
            if (this.ExistCountryCode || this.ExistAreaCode)
            {
                if (this.ExistCountryCode && this.ExistAreaCode)
                {
                    numberString = this.CountryCode + ' ' + this.AreaCode;
                }
                else if (this.ExistCountryCode)
                {
                    numberString = this.CountryCode;
                }
                else
                {
                    numberString = this.AreaCode;
                }
            }

            if (string.IsNullOrEmpty(numberString) && this.ExistContactNumber)
            {
                return this.ContactNumber;
            }
            else
            {
                numberString += ' ' + this.ContactNumber;
                return numberString;
            }
        }
    }
}
